File: typeorm/new-hire-onboard-self-help-portal/src/entity/TeamTodo.ts
import {Entity, PrimaryGeneratedColumn, ManyToOne, JoinColumn} from "typeorm";
import { Todo } from "./Todo";
import { Team } from "./Team";

@Entity({name: "team_todos"})
export class TeamTodo {

    @PrimaryGeneratedColumn()
    id: number;

    @ManyToOne(type => Todo, todo => todo.teamTodos)
    @JoinColumn({name: "todo_id"})
    todo: Todo

    @ManyToOne(type => Team, team => team.teamTodos)
    @JoinColumn({name: "team_id"})
    team: Team

}

File: spring/new-hire-onboard-self-help-portal/src/main/java/com/example/newhireonboardselfhelpportal/entities/TeamTodo.java
package com.example.newhireonboardselfhelpportal.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import com.fasterxml.jackson.annotation.JsonBackReference;

/**
 * TeamTodo
 */
@Entity(name = "team_todos")
public class TeamTodo {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name="todo_id")
    private Todo todo;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name="team_id")
    private Team team;


    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Todo getTodo() {
        return this.todo;
    }

    public void setTodo(Todo todo) {
        this.todo = todo;
    }

    public Team getTeam() {
        return this.team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }

}